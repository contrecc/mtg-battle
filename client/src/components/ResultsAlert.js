import React from 'react'
import { Alert } from 'reactstrap';

export default (props) => {
  const { winner, bothStrong, bothWeak } = props;
  let text = '';
  let color = '';

  if(bothStrong) {
    text = "You tied! Both creatures perish in brutal fashion, so we'll give you the point!";
    color = "secondary";
  } else if(winner) {
    text = "You chose correctly! Your creature demolished the other! Booyah!";
    color = "success";
  } else if(!winner) {
    text = "Ouch! Your creature got crushed!";
    color = "danger";
  } else if(bothWeak){
    text = "You tied! Both creatures are too weak to defeat the other, so we'll give you the point!";
    color = "secondary";
  }

  return (
    <Alert color={color}>
      {text}
      </Alert>
  )
}