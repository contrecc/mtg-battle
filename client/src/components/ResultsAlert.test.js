import ResultsAlert from './ResultsAlert';

describe('ResultsAlert', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ResultsAlert winner={true} bothStrong={false} bothWeak={false} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct text when user wins', () => {
    const wrapper = mount(
      <ResultsAlert winner={true} bothStrong={false} bothWeak={false} />
    );

    const div = wrapper.find('div');
    expect(div.hasClass('alert-success')).toBe(true);

    const text = div.text();
    expect(text).toEqual(
      'You chose correctly! Your creature demolished the other! Booyah!'
    );
  });

  it('renders the correct text when user loses', () => {
    const wrapper = mount(
      <ResultsAlert winner={false} bothStrong={false} bothWeak={false} />
    );

    const div = wrapper.find('div');
    expect(div.hasClass('alert-danger')).toBe(true);

    const text = div.text();
    expect(text).toEqual('Ouch! Your creature got crushed!');
  });

  it('renders the correct text both creatures are strong', () => {
    const wrapper = mount(
      <ResultsAlert winner={false} bothStrong={true} bothWeak={false} />
    );

    const div = wrapper.find('div');
    expect(div.hasClass('alert-secondary')).toBe(true);
    const text = div.text();

    expect(text).toEqual(
      "You tied! Both creatures perish in brutal fashion, so we'll give you the point!"
    );
  });

  it('renders the correct text both creatures are weak', () => {
    const wrapper = mount(
      <ResultsAlert winner={false} bothStrong={false} bothWeak={true} />
    );

    const div = wrapper.find('div');
    expect(div.hasClass('alert-secondary')).toBe(true);

    const text = div.text();
    expect(text).toEqual(
      "You tied! Both creatures are too weak to defeat the other, so we'll give you the point!"
    );
  });
});
