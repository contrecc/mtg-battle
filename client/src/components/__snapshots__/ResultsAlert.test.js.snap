// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsAlert renders correctly 1`] = `
<Alert
  closeAriaLabel="Close"
  color="success"
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  You chose correctly! Your creature demolished the other! Booyah!
</Alert>
`;
